### DISCUSSION TITLE
opendrive to commonroad and commonroad to lanelet2 conversion challenge

### DISCUSSION THREAD

**Discussion by nsaberi-geomate** [CommonRoad/commonroad-scenario-designer#6] [Q&A] [answered]
I am facing challenges in map conversions. When I run the module for # load OpenDRIVE file, parse it, and convert it to a CommonRoad scenario, I get this error:
scenario = opendrive_to_commonroad(input_path)

Traceback (most recent call last):
File “”, line 1, in 
File “XXX/commonroad-scenario-designer/crdesigner/map_conversion/map_conversion_interface.py”, line 94, in opendrive_to_commonroad
opendrive = parse_opendrive(input_file)
File “XXX/commonroad-scenario-designer/crdesigner/map_conversion/opendrive/opendrive_parser/parser.py”, line 39, in parse_opendrive
with file_path.open(“r”) as file_in:
AttributeError: ‘str’ object has no attribute ‘open’

I also tried this:

with open(“{}”.format(input_path), “r”) as file_in:
… opendrive = parse_opendrive(etree.parse(file_in).getroot())

But I am getting the same error.

Also tried the GUI version but it gives me no results even on sample files.

**Comment by smaierhofer** [comment, marked as answer]
It looks like the paths are of wrong type (string instead of pathlib.Path).
The input_path and output_path must be a [pathlib.Path ](https://docs.python.org/3/library/pathlib.html )object, e.g., as in our example [here](https://github.com/CommonRoad/commonroad-scenario-designer/blob/24a51c669fe4a5e426af86ef63c74424c5286ce6/tutorials/conversion_examples/example_opendrive_to_commonroad.py#L11).

**Comment by nsaberi-geomate** [comment]
Thanks for your response. Regarding input address, this command worked: 

crdesigner --input-file 'adress' --output-file 'address' odrcr 

However, for the conversion, I am facing another challenge regarding my opendrive config: 

XXX/anaconda3/envs/commonroad_env/lib/python3.9/site-packages/crdesigner/map_conversion/common/utils.py:65: UserWarning: geoidgrids removed from projection string
  warnings.warn("geoidgrids removed from projection string")
18-Jan-24 13:05:49 - WARNING - OpenDRIVE/traffic_signals.py: Unknown TrafficSignIDZamunda of ID -1 of subtype -1!
Traceback (most recent call last):

  File "XXX/anaconda3/envs/commonroad_env/bin/crdesigner", line 8, in <module>
    sys.exit(cli())

  File "XXX/anaconda3/envs/commonroad_env/lib/python3.9/site-packages/crdesigner/ui/command_line.py", line 97, in odrcr
    scenario = opendrive_to_commonroad(ctx.obj["input_file"])

  File "XXX/anaconda3/envs/commonroad_env/lib/python3.9/site-packages/crdesigner/map_conversion/map_conversion_interface.py", line 96, in opendrive_to_commonroad
    road_network.load_opendrive(opendrive)

  File "XXX/anaconda3/envs/commonroad_env/lib/python3.9/site-packages/crdesigner/map_conversion/opendrive/opendrive_conversion/network.py", line 123, in load_opendrive
    self._crosswalks.extend(get_crosswalks(road))

  File "XXX/anaconda3/envs/commonroad_env/lib/python3.9/site-packages/crdesigner/map_conversion/opendrive/opendrive_conversion/plane_elements/crosswalks.py", line 35, in get_crosswalks
    left_vertices = np.array([corners[1], corners[2]])

IndexError: index 1 is out of bounds for axis 0 with size 0

**Reply by smaierhofer** [reply]
this looks like a bug in the conversion of crosswalks.
A quick-fix could be to comment out the corresponding conversion part (in case you don't need crosswalks).

Can you provide the OpenDRIVE map?
Then, I would create an issue so that we can fix the bug in one of the next releases.

**Reply by nsaberi-geomate** [reply]
I did comment out line 123, but the same error is produced. Any thoughts? Can you provide your email address, please?

**Reply by smaierhofer** [reply]
I would try to comment out all calls to this function, i.e., all parts related to crosswalks. 
My email address: sebastian.maierhofer@tum.de

**Reply by smaierhofer** [reply]
@nsaberi-geomate The problems should be fixed on the develop branch.
